LandSandBoat/server
# [lua][sql] Colonization Reive Framework

**_I affirm:_**

## What does this pull request do?
Implements a basic framework to handle basic reive functionality such as spawning/clearing.
Defending mobs spawn with their corresponding reives.
TODOs are marked in code.

Note: This is a port from my server. It functions on my codebase and likely should function on upstream as well (Shift proofed), but I have not tested it yet until I get my retail client updated. Leaving as a draft until I or someone up to date else can test it.


## Steps to test these changes

1. Open scripts/globals/colonization_reive_data.lua. Pick a reive to go to. (The !pos xyz cords are listed for each reive).
2. Kill defender mobs. See that they respawn as expected matching the time listed in the colonization_reive_data data table.
3. Kill 1 reive objective mob. (Knotted Root etc). See that it does not respawn. You can lower the objective respawn time to test this.
4. Damage an objective but do not kill. Exit combat. Come back and see that the objective does not regenerate.
5. Kill all objectives. Reive ending should display correct animation effects and you should be able to pass the blocked area.

Note: Reive # 4 in Morimar Basalt Fields won't unblock. I added a TODO to it. It likely needs a capture for the blocker NPC/position.
